### ISSUE TITLE
expand multiple macros

### ISSUE THREAD

**Issue by cdacl** [lvjr/tabularray#77]
When I have more than one macro needed to expand in the table, and the macro including `&` or `\\` placed after the macro not including `&` or `\\`, the table broken.

This sample code is ok:

```
\def\a{a &b\\}
\def\b{z}


\begin{tblr}[expand={\a,\b}]{hlines,vlines}
 \b \\
 \a  
\end{tblr}

```

![1](https://user-images.githubusercontent.com/87848848/130041426-379b1c06-658c-4fc8-acb9-55c06fecc59d.png)

But when I changed the order of expand macroes, the table broken:
```
\begin{tblr}[expand={\b,\a}]{hlines,vlines}
 \b \\
 \a  
\end{tblr}
```
![2](https://user-images.githubusercontent.com/87848848/130041551-9b7e7338-60ac-4acf-92e9-e4699c332b7a.png)

**Comment by cdacl** [comment]
Addition, when both two macroes include `&` or `\\`, no matter what order it will make the table broken.

**Comment by lvjr** [comment]
In fact, `tabularray` can only expand **one** macro. You can not put multiple macros there.

**Comment by cdacl** [comment]
So, will more macros be expanded in futrue? :-)

**Comment by lvjr** [comment]
Yes.

**Comment by cdacl** [comment]
Thank you very much!

In fact, if all macros don't include `&` or `\\`, the table will be build correctly as expected.

**Comment by tobiasBora** [comment]
Also interested in this feature, especially if the macros can take arguments as inputs. See my issue here

**Comment by lvjr** [comment]
@tobiasBora It doesn't really help even if `tabularray` could expand multiple macros. For commands with optional arguments, you need exhaustive expansions, which is provided by `\expanded` command:
 
```tex
\documentclass{memoir}
\usepackage{xcolor}
\usepackage{tabularray}

\NewExpandableDocumentCommand{\yes}{O{Yes}m}{\SetCell{bg=green9}#1}
\NewExpandableDocumentCommand{\no}{O{No}m}{\SetCell{bg=red8}#1}

\begin{document}

\begin{tblr}[expand=\expanded]{cc}
  What I want               & is below              \\
  \SetCell{bg=green9} Yes   & \SetCell{bg=red8} No  \\
  \SetCell{bg=green9} Great & \SetCell{bg=red8} Bad \\
  What I get                & is below              \\
  \expanded{\yes{}}         & \expanded{\no{}}      \\
  \expanded{\yes[Great]{}}  & \expanded{\no[Bad]{}}
\end{tblr}

\end{document}
```

![image](https://user-images.githubusercontent.com/4239949/219223435-febfae09-29e1-4230-b1ba-46910633307f.png)

Note that you need to protect fragile commands (if any) inside them with `\unexpanded` command.

**Comment by tobiasBora** [comment]
Awesome trick, thanks a lot!

**Comment by mangkoran** [comment]
Hi @lvjr. I faced upon this issue too and wondering if this is the intended behavior of `tabularray`. I noticed @muzimuzhi already worked on multiple macros expansion, but we still need to declare the macros list to be expanded.

I mean, is this inability to do automatic macros expansion of `tabularray` is the intended behavior, or it will be worked out in the future to reach "behavior parity" with `tabular`?
